File: 044-backend/src/main/java/com/xiaou/entity/Booking.java
package com.xiaou.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("booking")
public class Booking {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String orderNo;
    private Long userId;
    private Long homestayId;
    private Long roomTypeId;
    private LocalDate checkInDate;
    private LocalDate checkOutDate;
    private Integer nights;
    private Integer guests;
    private BigDecimal totalPrice;
    private String contactName;
    private String contactPhone;
    private String remark;
    private Integer status;  // 0-待支付 1-待确认 2-已确认 3-进行中 4-已完成 5-已取消
    private String cancelReason;
    @TableLogic
    private Integer deleted;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: culture-travel/044-frontend/src/api/booking.ts
import request from '@/utils/request'

export function createBooking(data: {
  homestayId: number
  roomTypeId: number
  checkInDate: string
  checkOutDate: string
  guests?: number
  contactName: string
  contactPhone: string
  remark?: string
}) {
  return request.post('/booking/create', data)
}

export function cancelBooking(id: number, reason?: string) {
  return request.post(`/booking/cancel/${id}`, null, { params: { reason } })
}

export function getMyBookings(params: { current?: number; size?: number; status?: number }) {
  return request.get('/booking/my', { params })
}

export function getBookingDetail(id: number) {
  return request.get(`/booking/detail/${id}`)
}
